packages/tracing/src/idletransaction.ts

```diff
@@ -97,16 +102,19 @@ export class IdleTransaction extends Transaction {
       _idleHub.configureScope(scope => scope.setSpan(this));
     }
 
-    this._initTimeout = setTimeout(() => {
+    this._startIdleTimeout();
+    global.setTimeout(() => {
       if (!this._finished) {
+        this.setStatus('deadline_exceeded');
         this.finish();
       }
-    }, this._idleTimeout);
+    }, this._finalTimeout);
   }
 
   /** {@inheritDoc} */
   public finish(endTimestamp: number = timestampWithMs()): string | undefined {
     this._finished = true;
+    this._cancelIdleTimeout();
```

Comment:
Do we actually need to cancel here? `_startIdleTimeout` already checks for `_finished` before calling `finish()`.

Reply:
We do not, updated.